File: prisma/seed.ts
import { PrismaClient } from '@prisma/client'
import bcrypt from 'bcryptjs'
import crypto from 'crypto'

const prisma = new PrismaClient()

async function main() {
  console.log('Seeding database...')

  // Create admin user (admin@example.com / admin123)
  const hashedPassword = await bcrypt.hash('admin123', 10)

  const admin = await prisma.user.upsert({
    where: { email: 'admin@example.com' },
    update: {},
    create: {
      email: 'admin@example.com',
      nickname: '管理员',
      password: hashedPassword,
      role: 'admin',
      isActive: true,
    },
  })

  console.log('Admin user created:', admin.email)

  // Create demo app
  const demoAppId = `app_${crypto.randomBytes(8).toString('hex')}`
  const demoSecret = crypto.randomBytes(32).toString('hex')

  const app = await prisma.app.upsert({
    where: { appId: demoAppId },
    update: {},
    create: {
      appId: demoAppId,
      secret: demoSecret,
      name: '演示应用',
      domains: '',
      isActive: true,
    },
  })

  console.log('Demo app created:', app.appId)

  // Create demo regular user
  const demoPassword = await bcrypt.hash('user123', 10)
  await prisma.user.upsert({
    where: { email: 'user@example.com' },
    update: {},
    create: {
      email: 'user@example.com',
      nickname: '演示用户',
      password: demoPassword,
      role: 'user',
      isActive: true,
    },
  })

  console.log('Demo user created: user@example.com')
  console.log('Seeding complete!')
}

main()
  .catch((e) => {
    console.error(e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

File: webserver/app/seed.py
import asyncio
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from passlib.context import CryptContext
from webserver.app.database import engine, Base, async_session_factory
from webserver.app.models import User, App, ReviewBook, ReviewChapter

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")


async def seed():
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

    async with async_session_factory() as session:
        # Create admin user
        admin = User(
            email="admin@example.com",
            username="admin",
            nickname="Admin",
            password=pwd_context.hash("admin123"),
            role="admin",
            is_active=True,
        )
        session.add(admin)

        # Create test user
        test_user = User(
            email="test@example.com",
            username="testuser",
            nickname="Test User",
            password=pwd_context.hash("test123"),
            role="user",
            is_active=True,
        )
        session.add(test_user)

        # Create test app
        test_app = App(
            app_id="test-app-001",
            secret="test-secret-change-in-production",
            name="Test Application",
            domains="*",
            is_active=True,
        )
        session.add(test_app)

        # Create sample books and chapters
        book1 = ReviewBook(title="Sample Book 1", alias="sample-book-1")
        session.add(book1)
        await session.flush()

        chapter1 = ReviewChapter(
            book_id=book1.id,
            title="Chapter 1",
            alias="chapter-1",
            parents="",
        )
        session.add(chapter1)

        await session.commit()

    print("Seed data created successfully!")


if __name__ == "__main__":
    asyncio.run(seed())
